### ISSUE TITLE
support for multiple stylesheets on same component

### ISSUE THREAD

**Issue by moliver-bb** [jsstyles/react-jss#27]
Would be awesome to import multiple stylesheets. I want to keep my stylesheets in separate files and this would keep the reference to those files more clear when in the same component (instead of merging the two stylesheets together before importing). This would also allow the use of same style names in different files that are imported into the same component. 

Something like: 

```javascript
var React = require('react')
var useSheet = require('react-jss')

// You can use jss directly too!
var jss = require('jss')
var vendorPrefixer = require('jss-vendor-prefixer')
jss.use(vendorPrefixer)

var styles1 = require('./styles/styles1.js')
var styles2 = require('./styles/styles2.js')

var Button = React.createClass({
  render: function () {
    var styles1Classes = this.props.sheet.styles1.classes
    var styles2Classes = this.props.sheet.styles2.classes

    return (
  <div>
      <div className={styles1Classes.button}>
        <span className={styles1Classes.label}>
          {this.props.children}
        </span>
      </div>
      <div className={styles2Classes.button}>
        <span className={styles2Classes.label}>
          {this.props.children}
        </span>
      </div>
  </div>
    )
  }
})

module.exports = useSheets(Button, [{stylesheet: styles1, namespace: 'styles1'}, {stylesheet: styles2, namespace: 'styles2'}])
```

I can create a PR for this, but wanted to check with you on what you think the best implementation would be (creating a separate useSheets function or add it to the existing useSheet)

**Comment by kof** [comment]
What you are asking for is clear. Not clear is why. Most of the time the need for 2 sheets in one component indicates that you should split your component in 2.

**Comment by moliver-bb** [comment]
the main reason why is I want to keep a "general styles" stylesheet that I can then combine with more specific stylesheets when needed

**Comment by kof** [comment]
This sounds like you should extend your general styles with component specific or use parts from general styles in your specific one. But again its a theory. We need to see your code.

**Comment by moliver-bb** [comment]
yeah, we definitely can do it through extending. There are cases where we will be extending styles and not editing them at all. So the example would be:

```javascript

var Button = React.createClass({
  render: function () {
    var generalStyles = this.props.sheet.generalStyles.classes
    var buttonStyles = this.props.sheet.buttonStyles.classes

    return (
  <div className={generalStyles.flexBox}>
      <div className={buttonStyles.button}>
        <span className={buttonStyles.label}>
          {this.props.children}
        </span>
      </div>
  </div>
    )
  }
})

```

so in this case we could extend "generalStyles.flexBox" into the buttonStyles stylesheet, but we aren't really changing anything in those styles. So I was thinking it would be more straight forward to know that we are referencing the generalStyles sheet and not reference a buttonStyles sheet that in turn extended "flexBox" from generalStyles...

hopefully that explanation isn't too wordy. 

Also wanted to say thanks again for all this work!! react-jss has been an awesome tool for my projects!

**Comment by kof** [comment]
So in other words, you have 2 different concerns separated in 2 styles. One is about layout, another implements a button. Its totally valid to separate those 2 in different files.

I am not sure if its a good idea to have different multiple sheet refs for them instead of merging.

**Comment by kof** [comment]
Pro: if you are using totally independent sheets, they don't know which namespace is already used by another one and there can be collision.
Contra: One merged sheet is actually a higher level style sheet. It knows all props it wants to use. 

Also button sheet + button jsx sounds like it can be a separate button component.

**Comment by moliver-bb** [comment]
>pro: if you are using totally independent sheets, they don't know which namespace is already used by another one and there can be collision.

this is the main reason I would be using it. I guess my "flexBox" example isn't the best example. The use case might be: generalStyles.label is used in one part of the component and buttonStyles.label (which might or might-not extend generalStyles.label) is used in another section.

(the button was just used as an example I grabbed from the readme. That would normally be pulled out into its own component)

**Comment by kof** [comment]
Well, thats bad, we need real world use cases to fix. All examples pulled out of imaginations are not realistic enough.

**Comment by kof** [comment]
Also there is another way: If a sheet is used in lots of places, one can render this sheet directly using jss.createStyleSheet and pass it around where needed. The point of react-jss is ref-counting of component specific sheets.

**Comment by kof** [comment]
Also you can just import it from one file


styles.js

```javascript
export default {
  flexBox: {..}
}
```

sheet.js
```javascript
import jss from 'jss'
import styles from './styles'
export const sheet = jss.createStyleSheet(styles).attach()
```

```javascript
import {classes as generalClasses} from '../styles/sheet'

const Button = React.createClass({
  render: function () {
    const {classes} = this.props.sheet

    return (
  <div className={generalClasses.flexBox}>
      <div className={classes.button}>
        <span className={classes.label}>
          {this.props.children}
        </span>
      </div>
  </div>
    )
  }
})
```

**Comment by kof** [comment]
I have changed the description of this project in order to make its existence more clear.

**Comment by kof** [comment]
this could be a good way to get this #31
